### PR TITLE
OPM 157 write RFT data

### PR DESCRIPTION
**Pull request by chflo**
Added writing of eclipse RFT file to EclipseWriter. The writing of RFT data is handled in a separate class, EclipseWriteRFTHandler.

### CODE REVIEW

File: opm/core/io/eclipse/EclipseWriteRFTHandler.cpp [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+                                               std::vector<double>& swat,
+                                               std::vector<double>& sgas) {
+
+
+
+        std::vector<ecl_rft_node_type *> rft_nodes;
+        for (std::vector<WellConstPtr>::const_iterator ci = wells.begin(); ci != wells.end(); ++ci) {
+            WellConstPtr well = *ci;
+            if ((well->getRFTActive(simulatorTimer.currentStepNum())) || (well->getPLTActive(simulatorTimer.currentStepNum()))) {
+                ecl_rft_node_type * ecl_node = createEclRFTNode(well,
```
Comment (bska) [inline opm/core/io/eclipse/EclipseWriteRFTHandler.cpp]: Method `createEclRFTNode()` allocates memory for the `ecl_rft_node_type` pointer, presumably using `malloc()`.  What happens to that memory once `rft_nodes` goes out of scope?

Comment (chflo) [inline opm/core/io/eclipse/EclipseWriteRFTHandler.cpp]: The ERT library takes care of the cleanup of the ecl_rft_nodes that it receives in the ecl_rft_file_update method.
Cleanup of the ecl_rft_cells within each node are the responsibility of each ecl_rft_node, so this is also taken care of.

But I guess this deserves a comment in the code…I will add a comment on the cleanup of the ecl_rft_nodes

Comment (bska) [inline opm/core/io/eclipse/EclipseWriteRFTHandler.cpp]: > The ERT library takes care of the cleanup of the `ecl_rft_node`s that it receives in the `ecl_rft_file_update` method.

Aha.  I see.

> Cleanup of the `ecl_rft_cells` within each node are the responsibility of each `ecl_rft_node`, so this is also taken care of.

Good.

> But I guess this deserves a comment in the code…I will add a comment on the cleanup of the `ecl_rft_node`s

Yes please.  That'd be good.

**Comment by bska** [comment]
I take it all requisite upstream (i.e., opm-parser) changes have already been made, or does this depend on any as-yet unmerged PRs?

**Comment by chflo** [comment]
This PR is dependent on opm-parser PR's 431 and 434 which has both been merged

**Comment by bska** [comment]
> This PR is dependent on opm-parser PR's [...] which have both been merged

I see.  I'll do a test build and report back.  In the meantime, would you mind putting the test helper functions (`verifyRFTFile()` and the `create*()` family) into an anonymous namespace in `tests/test_EclipseWriteRFTHandler.cpp`?  That's the C++ equivalent of C's `static` functions and eliminates build diagnostics of the form
```
/home/bska/work/opm/src/opm/opm-core/tests/test_writenumwells.cpp: In function ‘Opm::DeckConstPtr createDeck(const std::string&)’:
/home/bska/work/opm/src/opm/opm-core/tests/test_writenumwells.cpp:136: warning: no previous declaration for ‘Opm::DeckConstPtr createDeck(const std::string&)’
```

**Comment by joakim-hove** [comment]
> or does this depend on any as-yet unmerged PRs?

It also depends on: https://github.com/Ensembles/ert/pull/712

Which was merged ~ two weeks ago.

**Comment by bska** [comment]
I didn't see any obvious changes to the output of opm-autodiff's [`flow`](https://github.com/OPM/opm-autodiff/blob/master/examples/flow.cpp) utility on the SPE9 benchmark.  Even the file sizes are the same and the files themselves compare equal in terms of `cmp(1)`.  Do I need to set some option to enable RFT output or do we need to make additional changes, e.g., to [`class Opm::BlackoilOutputWriter`](https://github.com/OPM/opm-autodiff/blob/master/opm/autodiff/SimulatorFullyImplicitBlackoilOutput.hpp) to benefit from this work?

Note: These are just incidental comments.  I will merge the patch set irrespective of the replies.  Probably on Monday (2015-03-09).

**Comment by joakim-hove** [comment]
You must ask for RFT with the RFT keyword in the SCHEDULE section. Then you
should get a CASE.RFT file with RFT output.
7. mars 2015 13:50 skrev "Bård Skaflestad" <notifications@github.com>
følgende:

**Comment by bska** [comment]
> You must ask for RFT with the `WRFT` keyword in the SCHEDULE section

Aha, that helps a lot.  Thanks.  I'll merge into master.

**Pull request merged**